src/lib/helpers.ts

```diff
@@ -69,3 +71,27 @@ export function* enumerate<T>(values: Iterable<T>, start = 0): Generator<[number
     yield [idx++, value];
   }
 }
+
+/**
+ * Return default value if selected value has different typeof
+ *
+ * @param value the selected value
+ * @param defaultValue the default to compare
+ */
+export function compareType(value: ValueType | ValueType[], defaultValue: ValueType): ValueType {
```

Comment:
I had trouble understanding what this function does due to its name.
To me, `compare` is supposed to return something saying that x is greater than y or the opposite.
Here, I have the impression it acts more like the `_.defaultTo` function of lodash/underscore : if the type is incompatible, then it return the default value, else it keep the actual value.

Reply:
Reading the next commit, may I suggest: `keepCurrentValueIfCompatibleType`? (it's a bit long, you'll probably find a better one ^^)

Reply:
keepCurrentValueIfCompatibleType seems great to me, i make the changes